biorxiv/word_vector_experiment/biorxiv_visualize_embeddings.py

```diff
@@ -0,0 +1,362 @@
+
+# ## UMAP Outliers
+
+# In[8]:
+
+
+g = (
+    p9.ggplot(biorxiv_umap_models_lastest[300])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.annotate("segment", x=10, y=7, xend=9.2, yend=5.2, arrow=p9.arrow(), color="red")
+    + p9.annotate("segment", x=10, y=-2.5, xend=12, yend=-0.6, arrow=p9.arrow(), color="red")
+    + p9.annotate("segment", x=7.5, y=-2.2, xend=6.2, yend=-2.2, arrow=p9.arrow(), color="red")
+    + p9.annotate("segment", x=7.5, y=1.5, xend=5.7, yend=1.5, arrow=p9.arrow(), color="red")
+    + p9.labs(
+        title="Outlier Detection of BioRxiv UMAP (300 dim)",
+        color="Article Category"
+    )
+)
+print(g)
+
+
+# In[9]:
+
+
+(
+    biorxiv_umap_models_lastest[300]
+    .query("umap1 > 5")
+)
+
+
+# Using the doi link (doi.org/doi url) so far the bioinformatics papers should be reclassified as Epigenetics (10.1101/290825) and Cancer biology (10.1101/599225). The animal behavior and cognition category (10.1101/066423 and 10.1101/045062) has shockingly similar titles, which means I found a duplicate of preprints. The articles have two different ids, but have the same authors and unsurprisingly close titles. Lastly, the genetics (10.1101/045666) preprint sounds like it should be a neuroscience preprint. So far it seems like the outliers are category errors/mysterious duplication.
+
+# In[10]:
+
+
+g = (
+    p9.ggplot(biorxiv_umap_models_lastest[300])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.annotate("segment", x=-5.5, y=5, xend=-5.6, yend=2.5, arrow=p9.arrow(), color="red")
+    + p9.annotate("segment", x=-4.8, y=5, xend=-4.8, yend=3.6, arrow=p9.arrow(), color="red")
+    + p9.labs(
+        title="Outlier Detection of BioRxiv UMAP (300 dim)",
+        color="Article Category"
+    )
+)
+print(g)
+
+
+# In[11]:
+
+
+(
+    biorxiv_umap_models_lastest[300]
+    .query("umap1 < -4.5")
+)
+
+
+# This article is mainly about evolutionary biology; however, it discusses intellegency quotient (IQ). This means the outlier is a combination of both fields; hence the point is in close proximity of the evolutionary biology section. The cancer biology preprint (10.1101/034132) seems like a biophysics paper or something to that nature.
+
+# In[12]:
+
+
+g = (
+    p9.ggplot(biorxiv_umap_models_lastest[300])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.annotate("rect", xmin=0, xmax=3, ymin=7.5, ymax=9.5, alpha = 0.4)
+    + p9.labs(
+        title="Outlier Detection of BioRxiv UMAP (300 dim)",
+        color="Article Category"
+    )
+)
+print(g)
+
+
+# In[13]:
```

Comment:
I wonder if it would be at all helpful to have multiple different datasets:
1. simple -- two categories that you expect to be pretty different, epidemiology and bioengineering
2. complex -- categories that are subtopics like cell biology, cancer biology, biochemistry..

I wonder if you can come up with a hierarchy of categories...not sure if this is useful..just a thought

Reply:
Hmm this is a good point. Given the scope of this PR I won't generate the plot here. Will keep it in mind though.